### ISSUE TITLE
DoubleTapped should not "eat" Tapped message

### ISSUE THREAD

**Issue by matwachich** [fyne-io/fyne#3901]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

The problem is that when extending a widget with Tappable and DoubleTappable, a Tapped event will not be caught immediatly, there will be a delay to check if it's not a DoubleTapped before calling Tapped.
This is problematic where a widget have to respond to the 2 events (example: file manager, where a Tapped=select, DoubleTapped=open/launch).

### Is it possible to construct a solution with the existing API?

Not without becoming very dirty by manualy handling doubleclick delays...

### Describe the solution you'd like to see.

The correct behaviour in my opinion is that every tapped event calls Tapped handler, after that, if a doubletapped is detected, also call DoubleTapped handler.

This is the case in [WinAPI for example](https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-lbuttondblclk).
`Double-clicking the left mouse button actually generates a sequence of four messages: [WM_LBUTTONDOWN](https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-lbuttondown), [WM_LBUTTONUP](https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-lbuttonup), WM_LBUTTONDBLCLK, and WM_LBUTTONUP.`

**Comment by dweymouth** [comment]
I ran into this same issue and ended up manually implementing DoubleTapped using the Tapped event for a component that I needed to react instantly to Tapped as well as have a DoubleTapped action.

Is this always the case though? Is there any use case where it makes sense for the Tapped event to happen after a delay to avoid sending Tapped if it ends up being a DoubleTapped?

**Comment by kpm** [comment]
It looks like it used to be that way until it was reported as a bug in 2020: https://github.com/fyne-io/fyne/issues/1275

If you try out almost any gui text editor (on Windows, Mac or even Linux), you'll notice the single click moves the cursor even if its part of a double click.  It would be nice if fyne did that by default on the desktop.  I noticed that once you implement the DoubleTapped event interface, the single tap response time feels really sluggish because it's waiting to see if its part of a double tap, which is the wrong behaviour.

In the mean time, I implemented it with the following, with a label for example (the delay is hardcoded):

```
type MyLabel struct {
    widget.Label
    OnDoubleTapped func()
    OnTapped       func()
    lastTapTime time.Time
}

func (d *MyLabel) Tapped(pe *fyne.PointEvent) {
    now := time.Now()

    // Check if this is a potential double tap
    if now.Sub(d.lastTapTime) < 300*time.Millisecond {
        if d.OnDoubleTapped != nil {
            d.OnDoubleTapped()
        } 
    } else {
        // Single tap - fire immediately
        if d.OnTapped != nil {
            d.OnTapped()
        }
    }

    d.lastTapTime = now
}
```

**Comment by andydotxyz** [comment]
> If you try out almost any gui text editor (on Windows, Mac or even Linux), you'll notice the single click moves the cursor even if its part of a double click.

This is true, but a text editor is not the only consumer of mouse events in a GUI toolkit - you are very welcome to add that to your own app, and even open a bug if our entry widget does not work that way (though I think it does!).
